tinygrad/renderer/assembly.py

```diff
@@ -133,14 +145,21 @@ def _cast(a, dtype:DType, atype:DType, bitcast=False, u=None, pred=False):
     uop,dtype,vin,args = u.uop,u.dtype,u.vin,u.arg
     if uop is UOps.IF:
       assert vin[0].dtype is not None
-      kk(*lang.render_bra(lb:=ssa_label('if', u), _cast(r[vin[0]], dtypes.bool, vin[0].dtype, u=u, pred=True), f"{lb}_true"), f"{lb}_true:")
+      kk(*lang.render_bra(ssa_label('if', u), _cast(r[vin[0]], dtypes.bool, vin[0].dtype, u=u, pred=True)))
     elif uop is UOps.BARRIER and lang.barrier: kk(lang.barrier)
     elif uop is UOps.ENDLOOP:
       kk(lang.asm_for_op[BinaryOps.ADD](r[vin[0]], r[vin[0]], "1", dtypes.int, lang.types[dtypes.int]),
           lang.asm_for_op[BinaryOps.CMPLT](pred:=ssa("pred", dtype="pred"), r[vin[0]], r[vin[0].vin[1]], dtypes.int, lang.types[dtypes.int]))
-      kk(*lang.render_bra(r_label[vin[0]], pred, f"{r_label[vin[0]]}_exit"), f"{r_label[vin[0]]}_exit:")
-    elif uop is UOps.ENDIF:
+      kk(*lang.render_bra(r_label[vin[0]], pred))
+    elif uop == UOps.ENDIF:
+      kk(f"@!{_cast(r[vin[0].vin[0]], dtypes.bool, vin[0].vin[0].dtype, u=u, pred=True)} bra {r_label[vin[0]]}_true;")
       kk(f"{r_label[vin[0]]}:")
+      if args:
+        if args[0].dtype.count > 1:
```

Comment:
Why does this need to be an if statement? You can always call scalar on the dtype.

Reply:
r[uop] contains a list if the count is bigger than 1 so we need to mov to all associated registers then